kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/DocumentServicesClientImpl.java

```diff
@@ -64,6 +64,9 @@ public DocumentInstance getDocument(String identifier) {
             ServiceResponse<DocumentInstance> response = (ServiceResponse<DocumentInstance>) executeJmsCommand( script, DescriptorCommand.class.getName(), "BPM" ).getResponses().get(0);
 
             throwExceptionOnFailure(response);
+            if (shouldReturnWithNullResponse(response)) {
+                return null;
```

Comment:
I wonder if this can be rewritten using Optional on master (7.0) - in general, returning null means there is a risk of NPE somewhere else. For 6.5 stream, though, I understand there is no perfect solution due to backward compatibility and support of JDK 6 and 7.